spec.html

```diff
@@ -36280,25 +36739,86 @@ <h1>
           <emu-alg>
             1. Let _basicWordChars_ be the CharSet containing every character in the ASCII word characters.
             1. Let _extraWordChars_ be the CharSet containing all characters _c_ such that _c_ is not in _basicWordChars_ but Canonicalize(_rer_, _c_) is in _basicWordChars_.
-            1. Assert: _extraWordChars_ is empty unless _rer_.[[Unicode]] is *true* and _rer_.[[IgnoreCase]] is *true*.
+            1. Assert: _extraWordChars_ is empty unless _rer_.[[EitherUnicode]] is *true* and _rer_.[[IgnoreCase]] is *true*.
             1. Return the union of _basicWordChars_ and _extraWordChars_.
           </emu-alg>
         </emu-clause>
 
+        <emu-clause id="sec-allcharacters" type="abstract operation">
+          <h1>
+            AllCharacters (
+              _rer_: a RegExp Record,
+            ): a CharSet
+          </h1>
+          <dl class="header">
+            <dt>description</dt>
+            <dd>Returns the set of “all characters” according to the regular expression flags.</dd>
+          </dl>
+          <emu-alg>
+            1. If _rer_.[[UnicodeSets]] is *true* and _rer_.[[IgnoreCase]] is *true*, then
+              1. Return the set of all Unicode code points _c_ that do not have a <a href="https://www.unicode.org/reports/tr44/#Simple_Case_Folding">Simple Case Folding</a> mapping (that is, scf(_c_)=_c_).
```

Comment:
`ecmarkup` does not like this line:

```
Error: spec.html:36759:60: could not find a preceding declaration for "c"
Error: spec.html:36759:193: could not find definition for scf
```

We could avoid the issue by simply stating

```
1. Return the set of all Unicode code points that do not have a <a href="https://www.unicode.org/reports/tr44/#Simple_Case_Folding">Simple Case Folding</a> mapping.
```

…but that is less clear. @tc39/ecma262-editors Is there a way to preserve the clarity while satisfying `ecmarkup`’s expectations?

Reply:
The `_c_` warning can be fixed either by rephrasing to "all Unicode code points _c_ such that _c_ does not have a..." (which ecmarkup will recognize as introducing a new alias) or by sticking `[declared="c"]` at the start of the step (which is the escape hatch for this error).

The `scf` one is trickier, since that really looks like it should be a call and ecmarkup expects calls to be of known things. The simplest fix is to wrap it in a non-functional HTML tag, for example `<emu-not-ref>scf</emu-not-ref>(_c_)`. Alternatively you could make it a link to something that defines what `scf` means when used in this way.

Reply:
Or, of course, you could make `scf` an actual defined thing - if you're going to use it in other places that probably makes sense. If you have a paragraph anywhere in the document which includes `<emu-eqn id="eqn-scf" aoid="scf">scf(_x_)</emu-eqn>` then ecmarkup will know what `scf` means (and will link to that place whenever the term is used).

Reply:
Thanks! I gave this a try.